File: app/src/main/java/com/vikingsen/cheesedemo/ux/cheesedetail/CheeseDetailModule.java
package com.vikingsen.cheesedemo.ux.cheesedetail;


import dagger.Module;
import dagger.Provides;

@Module
public class CheeseDetailModule {
    private final CheeseDetailContract.View view;

    CheeseDetailModule(CheeseDetailContract.View view) {
        this.view = view;
    }

    @Provides
    CheeseDetailContract.View provideView() {
        return view;
    }
}

File: app/src/main/java/com/vikingsen/cheesedemo/ux/cheesedetail/CheeseDetailModule.kt
package com.vikingsen.cheesedemo.ux.cheesedetail


import dagger.Module
import dagger.Provides

@Module
class CheeseDetailModule(private val view: CheeseDetailContract.View) {

    @Provides
    fun provideView(): CheeseDetailContract.View {
        return view
    }
}
